tools/RAiDER/utilFcns.py

```diff
@@ -532,3 +532,84 @@ def writeWeatherVars2HDF5(lat, lon, x, y, z, q, p, t, proj, outName=None):
         T = f.create_dataset('t', data=t)
 
         f.create_dataset('Projection', data=proj.to_json())
+
+
+
+# Part of the following UTM and WGS84 converter is borrowed from https://gist.github.com/twpayne/4409500
+# Credits go to Tom Payne
+
+_projections = {}
+
+
+def zone(coordinates):
+    if 56 <= coordinates[1] < 64 and 3 <= coordinates[0] < 12:
+        return 32
+    if 72 <= coordinates[1] < 84 and 0 <= coordinates[0] < 42:
+        if coordinates[0] < 9:
+            return 31
+        elif coordinates[0] < 21:
+            return 33
+        elif coordinates[0] < 33:
+            return 35
+        return 37
+    return int((coordinates[0] + 180) / 6) + 1
+
+
+def letter(coordinates):
+    return 'CDEFGHJKLMNPQRSTUVWXX'[int((coordinates[1] + 80) / 8)]
+
+
+def project(coordinates):
+    z = zone(coordinates)
+    l = letter(coordinates)
+    if z not in _projections:
+        _projections[z] = pyproj.Proj(proj='utm', zone=z, ellps='WGS84')
+    x, y = _projections[z](coordinates[0], coordinates[1])
+    if y < 0:
+        y += 10000000
+    return z, l, x, y
+
+
+def unproject(z, l, x, y):
+    if z not in _projections:
+        _projections[z] = pyproj.Proj(proj='utm', zone=z, ellps='WGS84')
+    if l < 'N':
+        y -= 10000000
+    lng, lat = _projections[z](x, y, inverse=True)
+    return (lng, lat)
+
+def WGS84_to_UTM(lon, lat):
+    shp = lat.shape
+    lon = np.ravel(lon)
+    lat = np.ravel(lat)
+    Z = lon.copy()
+    L = np.zeros(lon.shape,dtype='<U1')
+    X = lon.copy()
+    Y = lon.copy()
+    for ind in range(lon.__len__()):
+        longitude = lon[ind]
+        latitude = lat[ind]
+        z, l, x, y = project((longitude,latitude))
+        Z[ind] = z
+        L[ind] = l
+        X[ind] = x
```

Comment:
@leiyangleon 

Im not that familiar with UTM. But understadn from reading that its not a contineous XY system across zone boundaries. See http://geokov.com/education/utm.aspx 
` When working across zone boundaries multiple coordinate systems must be used. Coordinates in adjacent zones do not have a mathematical relationship.`

We provide a list of longitude and latitude and then project to UTM based on the lonlat.
If not continuous across zones then the xy coordinates might not be wrt to the same origin. e.g. what would be the different of using the UTM zone of the centroid and then computing the xy wrt to that zone?

Reply:
@dbekaert Got it. Will look into that next week.